R/conformal.R

```diff
@@ -0,0 +1,110 @@
+get_conformal_score <- function(x, score) {
+    response_name <- insight::find_response(attr(x, "model"))
+    response <- x[[response_name]]
+    if (!is.numeric(response) && score != "softmax") {
+      insight::format_error('The response must be numeric. Did you want to use `conformal_score="softmax"`?')
+    }
+    if (score == "residual_abs") {
+      out <- abs(response - x$estimate)
+    } else if (score == "residual_sq") {
+      out <- (response - x$estimate)^2
+    } else if(score == "softmax") {
+      model <- attr(x, "model")
+      response <- x[[insight::find_response(model)]]
+      if (is.numeric(response) && all(response %in% 0:1)) {
+        # See p.4 of Angelopoulos, Anastasios N., and Stephen Bates. 2022. “A
+        # Gentle Introduction to Conformal Prediction and Distribution-Free
+        # Uncertainty Quantification.” arXiv.
+        # https://doi.org/10.48550/arXiv.2107.07511.
+        # 1 minus the softmax output of the true class
+        out <- 1 - ifelse(response == 1, x$estimate, 1 - x$estimate)
```

Comment:
I think this could be simplified to `out <- ifelse(response != 1, x$estimate, 1 - x$estimate)`

Reply:
yep, or `response == 0` to be even more explicit. Thanks!